ios/FluentUI/Vnext/PersonaBadge/PersonaBadge.swift

```diff
@@ -0,0 +1,261 @@
+            avatarState.hasPointerInteraction = newValue
+        }
+    }
+
+    var hasRingInnerGap: Bool {
+        get {
+            return avatarState.hasRingInnerGap
+        }
+
+        set {
+            avatarState.hasRingInnerGap = newValue
+        }
+    }
+
+    var image: UIImage? {
+        get {
+            return avatarState.image
+        }
+
+        set {
+            avatarState.image = newValue
+        }
+    }
+
+    var imageBasedRingColor: UIImage? {
+        get {
+            return avatarState.imageBasedRingColor
+        }
+
+        set {
+            avatarState.imageBasedRingColor = newValue
+        }
+    }
+
+    var isOutOfOffice: Bool {
+        get {
+            return avatarState.isOutOfOffice
+        }
+
+        set {
+            avatarState.isOutOfOffice = newValue
+        }
+    }
+
+    var isRingVisible: Bool {
+        get {
+            return avatarState.isRingVisible
+        }
+
+        set {
+            avatarState.isRingVisible = newValue
+        }
+    }
+
+    var isTransparent: Bool {
+        get {
+            return avatarState.isTransparent
+        }
+
+        set {
+            avatarState.isTransparent = newValue
+        }
+    }
+
+    var presence: MSFAvatarPresence {
+        get {
+            return avatarState.presence
+        }
+
+        set {
+            avatarState.presence = newValue
+        }
+    }
+
+    var primaryText: String? {
+        get {
+            return avatarState.primaryText
+        }
+
+        set {
+            avatarState.primaryText = newValue
+        }
+    }
+
+    var ringColor: UIColor? {
+        get {
+            return avatarState.ringColor
+        }
+
+        set {
+            avatarState.ringColor = newValue
+        }
+    }
+
+    var secondaryText: String? {
+        get {
+            return avatarState.secondaryText
+        }
+
+        set {
+            avatarState.secondaryText = newValue
+        }
+    }
+
+    var size: MSFAvatarSize {
+        get {
+            return avatarState.size
+        }
+
+        set {
+            avatarState.size = newValue
+        }
+    }
+
+    var style: MSFAvatarStyle {
+        get {
+            return avatarState.style
+        }
+
+        set {
+            avatarState.style = newValue
+        }
+    }
+
+    init(size: MSFPersonaBadgeSize, avatarState: MSFAvatarState) {
+        self.badgeSize = size
+        self.avatarState = avatarState
+        self.tokens = MSFPersonaBadgeTokens(size: size)
+
+        super.init()
+    }
+}
+
+public struct PersonaBadge: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var tokens: MSFPersonaBadgeTokens
+    @ObservedObject var state: MSFPersonaBadgeViewStateImpl
+    let avatarView: AvatarView
+
+    public init(size: MSFPersonaBadgeSize) {
+        self.avatarView = AvatarView(style: .default, size: size.avatarSize)
```

Comment:
Following up on the discussion of having the Views in the State object or not (GitHub won't allow me to respond on the outdated thread :( ):

We can indeed discuss more about the pros and cons of doing that. And I think by the time the Collection is implemented, it would be ideal to have that pattern figured out as the collection will be dynamic and animations will have to follow on the state change of the collection of badges. 👍

Reply:
Sounds good!

That said, my ideal would be something like the following:
- The `Group` State tracks a list of View States, each of which is somehow uniquely identifiable
- At the View level of the Group, we declare PersonaBadge views for each View State. SwiftUI doesn't recreate Views unnecessarily because it doesn't need to: it can tell which View maps to which View State and does the right thing

That's dependent on being able to create View States independently from Views, and I don't believe that's something we've built yet. But it feels cleanest of all options.

Let's talk more offline - I'll draft up a simple proposal 🤓